### DISCUSSION TITLE
M2M, RESTAPI, when i create new Data. how to use the same data instead of creating a new one

### DISCUSSION THREAD

**Discussion by NMTuan** [directus/directus#14207] [Archive]
I created two collection: 
news and tags. news.tags <- M2M -> tags.name

I create a new data with restapi:
post /items/news
body {title: 'first news', tags: [{name: 'art'}]}

and I create another data( without know art tags id):

post /items/news
body {title: 'test news', tags: [{name: 'art}]}

now the are two same data in the tags collection.

how to use the same data instead of creating a new one, without know tag.id

**Comment by br41nslug** [comment]
You'll need to provide the primary key if you want to update a related record instead of creating a new one. You could either make the tags.name the primary key then this should work without change or fetch the tags `id` first or create the link after the fact. https://docs.directus.io/reference/introduction/#one-to-many-many-to-many

**Reply by NMTuan** [reply]
it worked. thank you
